app/models/generic_object_definition.rb

```diff
@@ -0,0 +1,12 @@
+class GenericObjectDefinition < ApplicationRecord
+  validates :name, :presence => true, :uniqueness => true
+
+  serialize :properties, Hash
```

Comment:
Should there be some validations on the contents of properties?

Would it be better to have accessors for the things that will be stored in properties?  That is, you already have defined_attributes...I'm wondering if there should be a `defined_attributes=`.  What other things will be stored in properties?